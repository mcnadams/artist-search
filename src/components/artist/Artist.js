import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function Artist({ artist }) {
  const nameURI = encodeURIComponent(artist.name);
  const path = `/artist/${nameURI}/${artist.id}`;
  return (
    <>
      <Link to={path} style={{ 
        color: 'black', 
        padding: '5px', 
        fontSize: '1em' 
      }}>{artist.name}</Link>
    </>
  );
}

Artist.propTypes = {
  artist: PropTypes.object.isRequired
};

export default Artist;
