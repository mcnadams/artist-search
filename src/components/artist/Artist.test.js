import React from 'react';
import { shallow } from 'enzyme';
import Artist from './Artist';

describe('Artist renders', () => {
  it('renders Artist', () => {
    const wrapper = shallow(<Artist artist={{ name: 'lady', id: '5' }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
