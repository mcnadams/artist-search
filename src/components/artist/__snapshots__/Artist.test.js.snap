// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist renders renders Artist 1`] = `
<Fragment>
  <Link
    style={
      Object {
        "color": "black",
        "fontSize": "1em",
        "padding": "5px",
      }
    }
    to="/artist/lady/5"
  >
    lady
  </Link>
</Fragment>
`;
